src/structured_data.rs

```diff
@@ -96,6 +96,29 @@ impl StructuredData {
         DataIdentifier::Structured(self.name, self.type_tag)
     }
 
+    /// Delete self
+    pub fn delete_self_against_successor(&mut self,
+                                         other: &StructuredData)
+                                         -> Result<(), RoutingError> {
+        match self.validate_self_against_successor(other) {
```

Comment:
I'm thinking of a scenario like this: A client `Post`s a change to the structured data, and the attacker controls the relay node. They could then drop the message and send a `Delete` containing the same data (with valid signatures and which is a valid successor).

In either case, it wouldn't hurt to require the new data version to be empty, would it?